common/config/azure-pipelines/templates/docs-build.yaml

```diff
@@ -0,0 +1,54 @@
+# The template defines the primary steps for creating the TypeDoc API files, gathering other md files,
+# and running a step to combine everything.
+
+parameters:
+  - name: workingDir
+    type: string
+    default: $(System.DefaultWorkingDirectory)
+  - name: stagingDir
+    type: string
+    default: $(System.ArtifactsDirectory)/staging
+  - name: outputDir
+    type: string
+    default: $(Agent.BuildDirectory)/tempDocsBuild/public_build
+  - name: downloadCurrentBuildArtifacts
+    type: boolean
+    default: false
+
+steps:
+- script: node common/scripts/install-run-rush.js install
+  displayName: rush install
+  workingDirectory: ${{ parameters.workingDir }}
+
+- script: node common/scripts/install-run-rush.js build -v
+  displayName: rush build
+  workingDirectory: ${{ parameters.workingDir }}
+
+- script: node common/scripts/install-run-rush.js docs

```

Comment:
I've also always thought that we should be able to: 
- Parse rush.json
- Make a list of the packages with `shouldPublish=true`
- Pass that list to `rush build` and `rush docs` with `-t` option
 
Should speed up the builds a bit by not building and running Typedoc on undocumented/unpublished packages (test apps and test packages come to mind)

Reply:
We could use the `--to-version-policy` and that would only build the published packages however we still have things like "example-code" and some other that contribute to the docs site as well.

We could probably do something better though here.

Reply:
Something like `rush build --to-version-policy prerelease-monorepo-lockStep && rush build --to example-code-snippets && rush build --to example-code-app`.

Although that being said I do want to clean up our usage of example code.

Reply:
Oh, nice. So the Rush team already thought of this use case. I'd be in favor of your suggestion. Every half a minute counts.

Reply:
Cool, let me test out that change.

Reply:
Same for `rush docs`. I'll push that.

Reply:
@naveedkhan8067 did you create a new pipeline to run this build? I don't see it in the previous section.

Reply:
Nevermind. Rush docs isn't as smart as rush build, and most of those "extra" packages don't have a docs script anyway, whoops.

Reply:
Saved 2.5 mins from the previous `rush compile`